tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -620,6 +632,147 @@ def test__get_public_collection_verify_body_is_reshaped_correctly__OK(self):
+            self.assertEqual(revision.version_id.id, resp_collection["id"])
+            self.assertIsNone(resp_collection["datasets"][0]["revision_of"])
+            self.assertIsNone(resp_collection["datasets"][0]["revised_at"])
+            self.assertEqual(0, resp_collection["datasets"][0]["revision"])
+            self.assertEqual(revision.datasets[0].version_id.id, resp_collection["datasets"][0]["id"])
+            self.assertIn(revision.datasets[0].version_id.id, resp_collection["datasets"][0]["explorer_url"])
+
+        revised_dataset = self.generate_dataset(
+            collection_version=revision, replace_dataset_version_id=revision.datasets[0].version_id
+        )
+        with self.subTest("get unpublished version with replace dataset and published version"):
+            resp_collection = _test_responses_are_equal([revision.version_id], revision.version_id, all_headers)
+            self.assertEqual("PRIVATE", resp_collection["visibility"])
+            self.assertIsNone(resp_collection.get("revising_in"))
+            self.assertEqual(revision.collection_id.id, resp_collection["revision_of"])
+            self.assertIsNone(resp_collection["revised_at"])
+            self.assertIsNotNone(resp_collection["published_at"])
+            self.assertTrue(resp_collection["collection_url"].endswith(revision.version_id.id))
+            self.assertEqual(revision.version_id.id, resp_collection["id"])
+            self.assertEqual(published.datasets[0].dataset_id.id, resp_collection["datasets"][0]["revision_of"])
+            self.assertIsNotNone(resp_collection["datasets"][0]["revised_at"])
+            self.assertEqual(0, resp_collection["datasets"][0]["revision"])
+            self.assertEqual(revised_dataset.dataset_version_id, resp_collection["datasets"][0]["id"])
+            self.assertIn(revised_dataset.dataset_version_id, resp_collection["datasets"][0]["explorer_url"])
+
+        self.business_logic.publish_collection_version(revision.version_id)
+        with self.subTest("get updated published version"):
+            resp_collection = _test_responses_are_equal(
+                [published.collection_id, revision.version_id], published.collection_id, all_headers
+            )
+            self.assertEqual("PUBLIC", resp_collection["visibility"])
+            self.assertIsNone(resp_collection.get("revising_in"))
+            self.assertIsNone(resp_collection["revision_of"])
+            self.assertIsNotNone(resp_collection["revised_at"])
+            self.assertIsNotNone(resp_collection["published_at"])
+            self.assertTrue(resp_collection["collection_url"].endswith(published.collection_id.id))
+            self.assertEqual(published.collection_id.id, resp_collection["id"])
+            self.assertIsNone(resp_collection["datasets"][0]["revision_of"])
+            self.assertIsNotNone(resp_collection["datasets"][0]["revised_at"])
+            self.assertEqual(0, resp_collection["datasets"][0]["revision"])
```

Comment:
Should we make updating the # of revisions as part of this PR as well since we're introducing the testing? Its hardcoded to 0 now for all collections. I think we can figure it out by querying the # of CollectionVersions (I believe we drop deleted revisions that do not get published), but let's double check that

Also happy to say its out of scope for this PR and do it separately

Reply:
I'd say out of scope. Fixing revised_at is a higher priority than fixing revision count.